test/crypto_subtle_test.dart

```diff
@@ -0,0 +1,537 @@
+      expect(
+        () => subtle.window.crypto.getRandomValues(Uint8List(1000000).toJS),
+        throwsA(
+          isA<subtle.JSDomException>()
+              .having(
+                (e) => e.name,
+                'name',
+                'QuotaExceededError',
+              )
+              .having(
+                (e) => e.message,
+                'message',
+                "Failed to execute 'getRandomValues' on 'Crypto': The ArrayBufferView's byte length (1000000) exceeds the number of bytes of entropy available via this API (65536).",
+              ),
+        ),
+      );
+    });
+
+    test(testOn: 'firefox', 'getRandomValues: too long', () {
+      expect(
+        () => subtle.window.crypto.getRandomValues(Uint8List(1000000).toJS),
+        throwsA(
+          isA<subtle.JSDomException>()
+              .having(
+                (e) => e.name,
+                'name',
+                'QuotaExceededError',
+              )
+              .having(
+                (e) => e.message,
+                'message',
+                'Crypto.getRandomValues: getRandomValues can only generate maximum 65536 bytes',
+              ),
+        ),
+      );
+    });
+
+    test(testOn: 'safari', 'getRandomValues: too long', () {
+      expect(
+        () => subtle.window.crypto.getRandomValues(Uint8List(1000000).toJS),
+        throwsA(
+          isA<subtle.JSDomException>()
+              .having(
+                (e) => e.name,
+                'name',
+                'QuotaExceededError',
+              )
+              .having(
+                (e) => e.message,
+                'message',
+                'The quota has been exceeded.',
+              ),
+        ),
+      );
+    });
+
+    test(testOn: 'chrome', 'getRandomValues: not supported type', () {
+      expect(
+        () => subtle.window.crypto.getRandomValues(Float32List(32).toJS),
+        throwsA(
+          isA<subtle.JSDomException>()
+              .having(
+                (e) => e.name,
+                'name',
+                'TypeMismatchError',
+              )
+              .having(
+                (e) => e.message,
+                'message',
+                "Failed to execute 'getRandomValues' on 'Crypto': The provided ArrayBufferView is of type 'Float32', which is not an integer array type.",
+              ),
+        ),
+      );
+    });
+
+    test(testOn: 'firefox', 'getRandomValues: not supported type', () {
+      expect(
+        () => subtle.window.crypto.getRandomValues(Float32List(32).toJS),
+        throwsA(
+          isA<subtle.JSDomException>()
+              .having(
+                (e) => e.name,
+                'name',
+                'TypeMismatchError',
+              )
+              .having(
+                (e) => e.message,
+                'message',
+                'The type of an object is incompatible with the expected type of the parameter associated to the object',
```

Comment:
I do wonder, if perhaps we'd be better off only testing `e.name` and not testing `e.message`.

It's not really part of the specification what the message should be, and I don't think we'll benefit from having test cases that break whenever Chrome or Firefox is updated and starts producing better error messages.

But ideally, the kind of exception they throw `e.name` should be somewhat stable.

Reply:
@jonasfj 
Hi, I pushed [ba1ff2e](https://github.com/google/webcrypto.dart/pull/88/commits/ba1ff2ec9df2af55bf1af0a7cbf945ee0c0aa1d0) to simplify test cases. Could you review it again? Thanks.